src/lightning/pytorch/core/module.py

```diff
@@ -648,21 +650,32 @@ def forward(self, *args: Any, **kwargs: Any) -> Any:
         """
         return super().forward(*args, **kwargs)
 
-    def training_step(self, *args: Any, **kwargs: Any) -> STEP_OUTPUT:  # type: ignore[return-value]
+    @overload
+    def training_step(self, batch: Any, batch_idx: int) -> Optional[STEP_OUTPUT]:
```

Comment:
Could we make the Optional as part of the `STEP_OUTPUT` like before? This would make it easier to delete. For users who use auto-complete, I believe most of them want no type hints.

Reply:
You mean modify the definition of `STEP_OUTPUT` in `src/lightning/pytorch/utilities/types.py`?